### DISCUSSION TITLE
What versions of yarn/npm does Renovate bot support best?

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#12581] [Request Help] [answered]
A repository I help maintain uses Yarn v1, which doesn't get many updates anymore, so we need to upgrade or migrate to `npm`. I'm trying to figure out what package manager - and which version - would work best with Renovate bot:

- Yarn 2 [^yarn-meta-not-updated]
- Yarn 3 [^yarn-issue] [^patch-clobber]
- npm 7 [^npm7-remediation]
- npm 8 [^engine-strict-true-bug]

To try to figure this out myself, I've searched the repository for open issues. I found a open issue for each of these managers, so that's not really pointing the way for me.

Do you have any advice on which package manager to pick?

[^yarn-meta-not-updated]: https://github.com/renovatebot/renovate/issues/9516
[^yarn-issue]: https://github.com/renovatebot/renovate/issues/11985
[^npm7-remediation]: https://github.com/renovatebot/renovate/issues/10371
[^patch-clobber]: https://github.com/renovatebot/renovate/issues/9509
[^engine-strict-true-bug]: https://github.com/renovatebot/renovate/issues/12068

**Comment by rarkins** [comment, marked as answer]
I think you'd want either Yarn 3 or npm 8, although I'm still happy with Yarn 1 in this repo. I don't think the issues raised for npm 7-8 should worry you.

**Reply by HonkingGoose** [reply]
Thank you for the quick answer!

Do you want me to create a new issue to track upgrading/migrating the Yarn 1 version in our repo?

**Reply by rarkins** [reply]
No, I think let's leave as-is for now

**Reply by HonkingGoose** [reply]
Okay, I'll leave it as-is for now then. 😉
